### PR TITLE
endpoints: only delete endpoint when the service error is not found

### PR DESCRIPTION
**Pull request by mfojtik**
**What this PR does / why we need it**:

In case the error returned from the Get() is not NotFound, we should not delete the endpoint as we are not sure if the service exists or not. 
**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/endpoint/endpoints_controller.go [hunk 1]
```diff
@@ -393,17 +393,18 @@ func (e *EndpointController) syncService(key string) error {
 		return err
 	}
 	service, err := e.serviceLister.Services(namespace).Get(name)
-	if err != nil {
+	if err != nil && errors.IsNotFound(err) {
```
Comment (mfojtik) [inline pkg/controller/endpoint/endpoints_controller.go]: @deads2k i'm actually not sure if a Lister can ever return something else than 404 as error...

Comment (sttts) [reply pkg/controller/endpoint/endpoints_controller.go]: Should we check that we actually own the endpoint before deleting? If it is an external service, we don't.

Comment (mfojtik) [reply pkg/controller/endpoint/endpoints_controller.go]: @sttts if endpoints have ownerRefs we don't have to do this at all I guess (GC should remove endpoints).

Comment (sttts) [reply pkg/controller/endpoint/endpoints_controller.go]: right, even better.

**Pull request closed without merging**